src/path_rewriting.rs

```diff
@@ -97,6 +99,30 @@ fn get_abs_path(source_dir: &Option<PathBuf>, rel_path: PathBuf) -> (PathBuf, Pa
     (abs_path, rel_path)
 }
 
+fn reverse(mut s: String) -> String {
+    unsafe {
+        let vec = s.as_mut_vec();
+        vec.reverse();
+    }
+    s
+}
+
+fn map_partial_path(files_tst: &TSTMap<String>, mut path: PathBuf) -> PathBuf {
+    // No need to map the path if it exists.
+    if path.exists() {
+        return path;
+    }
+
+    let mut found = false;
+    for (_key, value) in files_tst.prefix_iter(&reverse(path.to_str().unwrap().to_string())) {
+        assert!(!found);
```

Comment:
if we've tests/class/main.cpp and tests/foo/class/main.cpp and the path is "class/main.cpp", then it will fail here.
Is such situation possible ?